### PR TITLE
Fix deleting children fixes #112

### LINKED ISSUE #112: Deleting a children from a choral makes the choral unaccessable
If you created a choral "CHORAL1" that has say "CHILDREN1" and you delete this children, when trying to view the chart for CHORAL1 you will get "TypeError: Cannot read property 'choralId' of null"

### PR DESCRIPTION
**Pull request by james-woo**

### CODE REVIEW

**Review by andreirtaylor — COMMENTED** [review]
Choral.update({ children: choral._id },

Shouldn't this be 

Choral.update({},

Don't we want to get the children from all the chorals

**Comment by james-woo** [comment]
update( {QUERY}, {UPDATE})
So the query is looking for all chorals with that as a child.

**Comment by andreirtaylor** [comment]
You can query arrays like that? I thought you had to do $in

**Comment by james-woo** [comment]
Yep! That is the magic of mongo

**Comment by andreirtaylor** [comment]
I mean I guess this works for now. But the way we have implemented this is really a house of cards. Removing the children doesn't mean that the choral will work as intended just that it will continue to show values. If everyone understands that lgtm.

**Comment by james-woo** [comment]
How is that so? It removes dangling children pointers. When a user deletes a choral, the request also removes itself from any of its parents. I am not sure what you really mean.

**Comment by andreirtaylor** [comment]
The choral function can depend on the children. If the child isn't there it
potentially could cause a error.

**Comment by cpjk** [comment]
:eyes:

**Comment by cpjk** [comment]
> The choral function can depend on the children. If the child isn't there it
potentially could cause a error.

I guess for now, assume the user knows that

File: routes/chorals.js [hunk 1]
```diff
@@ -323,8 +323,18 @@ router.delete('/:choralId', function(req, res, next) {
         return res.send('500'); // notify client of failure
       }
 
-      res.flash('success', 'Choral successfully deleted.');
-      return res.send('204');  // successful deletion
+      // Delete all children choral from parents
```
Comment (cpjk) [inline routes/chorals.js]: I think it would make more sense to say "Delete this choral from any parents it has"

**Comment by cpjk** [comment]
1 comment then i think G2G

**Pull request merged**